### PR TITLE
Implemented React component for dashboard API

### LINKED ISSUE #52: Student dashboard: UI for View courses and course enrollments
As a student, I need to be able to see which courses are included in my micromasters programs, and understand whether I'm enrolled. 

The list of courses in a particular MM program are available from the MM API. Student enrollments are available from the edX enrollment API. The enrollment API returns a list of the courses the student is enrolled in and whether she is ID verified. 

This story covers display the list of courses, links to the course home page (aka tab) and an indicator as to whether the student is enrolled, enrolled & verified or not enrolled.

### LINKED ISSUE #123: Display user enrollments in dashboard
Depends on #125

Depending on the course status returned from the student dashboard api, display the appropriate data: i.e. a string, a button or a grade. 

See https://projects.invisionapp.com/share/KR6FYCZTX#/screens/142338434

### PR DESCRIPTION
**Pull request by noisecapella**
#### What are the relevant tickets?
Fixes #123

#### What's this PR do?
Updates the `CourseList` component to show the program tables like they are displayed in the UI. There are tests for each case of course status. Currently there is fake data for the dashboard API and real data for the course and program API so you may not be able to get the two to mix well to see the results when testing.

#### How should this be manually tested?
Create some programs, mark all of them live, then look at the dashboard

#### Any background context you want to provide?
static/js/util/util.js

This contains the code which maps the course status to the UI to display

#### Screenshots (if appropriate)
screenshot from 2016-04-05 18-34-18

### CODE REVIEW

**Comment by noisecapella** [comment]
Still needs to link the courses from the dashboard API to the programs in the programs API so we can group the courses and display tables logically

File: static/js/components/VerifyButton.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+import Button from 'react-bootstrap/lib/Button';
+
+class VerifyButton extends React.Component {
+  render() {
+    return <Button bsStyle="success">VERIFY</Button>;
```
Comment (pdpinch) [inline static/js/components/VerifyButton.js]: Let's use the edX language here, if not their styling

`Upgrade to Verified`

File: static/js/components/CourseList.js [hunk 2]
```diff
@@ -1,19 +1,50 @@
...
+    let tables = courseList.programList.map(program => {
+      let courses = courseList.courseList.
+        filter(course => course.program === program.id).
+        map(course => dashboardLookup[course.id]).
+        filter(course => course);
+
+      return <div key={program.id}>
+        <ul className="course-list-header">
+          <li>{program.title}</li>
+          <li />
```
Comment (aliceriot) [inline static/js/components/CourseList.js]: empty `<li>`- does this get filled in with something later?

Comment (noisecapella) [inline static/js/components/CourseList.js]: It won't be filled with anything, but we use it as a spaceholder to match the `li` elements in the table below. They're laid out with bourbon-neat which does CSS positioning.

File: static/js/components/CourseList.js [hunk 3]
```diff
@@ -1,19 +1,50 @@
...
 class CourseList extends React.Component {
   render() {
-    const { courseList } = this.props;
+    const { dashboard, courseList } = this.props;
+    let dashboardLookup = {};
+    for (let course of dashboard.courses) {
+      dashboardLookup[course.id] = course;
+    }
+
+    let tables = courseList.programList.map(program => {
```
Comment (aliceriot) [inline static/js/components/CourseList.js]: is the `programList` here the same as the `programList` that's dispatched to the store after we use the `fetchCourseList` method in `actions/index.js`?

Comment (noisecapella) [inline static/js/components/CourseList.js]: Yes. The naming could use some work :)

File: static/js/components/CourseList.js (hunk 2 above)
Comment (aliceriot) [inline static/js/components/CourseList.js]: ah, cool, I figured it was some sort of spacing thing

File: static/js/components/VerifyButton.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+class VerifyButton extends React.Component {
+  render() {
+    return <Button bsStyle="success">UPGRADE TO VERIFIED</Button>;
+  }
+}
+
+VerifyButton.PropTypes = {
+};
+
+export default VerifyButton;
```
Comment (aliceriot) [inline static/js/components/VerifyButton.js]: for this component (and the above component) are we anticipating needing to add more functionality to the buttons themselves?

Comment (noisecapella) [inline static/js/components/VerifyButton.js]: Probably not, it was a convenience for testing but then I changed the way these are tested. I'll inline this and the EnrollButton at the point where it's used

Comment (aliceriot) [inline static/js/components/VerifyButton.js]: :100:

File: static/js/constants.js [hunk 5]
```diff
@@ -1,30 +1,90 @@
...
-      "id": 7,
-      "title": "Course plTYyFhMcEBj",
-      "enrollment_start": null,
-      "start_date": null,
-      "enrollment_url": null,
-      "prerequisites": null,
-      "program": 3
-    }
-  ];
+export const COURSE_LIST_RESPONSE = [
```
Comment (aliceriot) [inline static/js/constants.js]: is this stuff all just for testing purposes?

**Comment by aliceriot** [comment]
back to you! it all looks good to me, excepting the few changes (inlining the buttons) we talked about.

File: static/js/constants.js (hunk 5 above)
Comment (noisecapella) [inline static/js/constants.js]: Yes, we are also using this fake data in the API for now, since we don't have the real REST API available yet

**Comment by noisecapella** [comment]
Ok, ready for review again. I squashed due to merging @annagav's changes

**Comment by aliceriot** [comment]
@noisecapella could you point to what has changed since the last review? I see deleting the `<button>` utility classes and some changes to accomodate rebasing over [](https://github.com/mitodl/micromasters/commit/32a212b32c210e66beb11a58b9e3ad38231a8680).

if that's it I think it's a :+1: from me!

**Comment by noisecapella** [comment]
Yes that's correct, I'll go ahead and merge

**Pull request merged**